src/main/scala/PSO/PSOSystemFactory.scala

```diff
@@ -21,22 +21,24 @@ package pso
 import akka.actor.{ActorSystem, Props}
 
 class PSOSystemFactory(config: PSOConfiguration) {
-  val problemType = config.problemType
-  val solutions = config.initialSolutions
-  val topology = config.topology
+  private val problemType = config.problemType
+  private val solutions = config.initialSolutions
+  private val topology = config.topology
   private def obj = config.objectiveFunction
 
-  def build(randomSeed: Option[Int] = None): PSOSystem = {
-    val system = ActorSystem("SwarmSystem")
-    val master = system.actorOf(Props(
-      classOf[SwarmMaster],
-      obj,
-      config.problemType
-    ))
-    val r = randomSeed match {
-      case None => new scala.util.Random(System.currentTimeMillis)
-      case Some(seed) => new scala.util.Random(seed)
+  def build(randomSeed: Option[Int] = None,
+    actorSystem: Option[ActorSystem] = None): PSOSystem = {
+    val system = actorSystem match {
```

Comment:
You could clean this up using a getOrElse.

actorSystem.getOrElse(ActorSystem("PSOSwarmSystem")